test/test_utils/sagemaker.py

```diff
@@ -139,10 +140,12 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
     test_report = os.path.join(os.getcwd(), "test", f"{job_type}_{tag}.xml")
     local_test_report = os.path.join(UBUNTU_HOME_DIR, "test", f"{job_type}_{tag}_sm_local.xml")
 
-
-    efa_flag = ""
-    efa_dedicated = os.getenv("EFA_DEDICATED", "False").lower() == "true"
-    efa_flag = '--efa' if efa_dedicated else '-m \"not efa\"'
+    if is_pr_context():
```

Comment:
could you provide comments on why there is a diff experience in PR vs otherwise?

Reply:
Sure. Adding comments now.